apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js

```diff
@@ -13,10 +14,42 @@ import {
   selectAssessmentInstanceLog,
   selectAssessmentInstanceLogCursor,
 } from '../../lib/assessment';
+import { InstanceQuestionSchema, IdSchema } from '../../lib/db-types';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
+export const AssessementInstanceStatsSchema = z.object({
+  assessment_instance_id: IdSchema,
+  average_submission_score: z.number().nullable(),
+  client_fingerprint_id_change_count: z.number(),
+  first_submission_score: z.number().nullable(),
+  incremental_submission_points_array: z.array(z.number()).nullable(),
+  incremental_submission_score_array: z.array(z.number()).nullable(),
+  instance_question_id: IdSchema,
+  last_submission_score: z.number().nullable(),
+  max_submission_score: z.number().nullable(),
+  number: z.string().nullable(),
+  qid: z.string(),
+  question_id: IdSchema,
+  some_nonzero_submission: z.boolean().nullable(),
+  some_perfect_submission: z.boolean().nullable(),
+  some_submission: z.boolean().nullable(),
+  submission_score_array: z.array(z.number()).nullable(),
+  title: z.string().nullable(),
+});
+
+const DateDurationResultSchema = z.object({
+  assessment_instance_date_formatted: z.string(),
+  assessment_instance_duration: z.string(),
+});
+
+const InstanceQuestionRowsSchema = InstanceQuestionSchema.extend({
```

Comment:
Nit: the base noun in schemas should typically be singular since this is the schema for a particular row.

```suggestion
const InstanceQuestionRowSchema = InstanceQuestionSchema.extend({
```

Reply:
@mylesw27 note that you'd need to update this name elsewhere in the file too.

Reply:
Updated this name in commit 254ecba.